pkg/command/root_test.go

```diff
@@ -908,3 +903,311 @@ func TestCompletionShortHelpInActiveHelp(t *testing.T) {
+		pi.SupportedContextType = p.supportedContextType
+	}
+	if len(p.aliases) != 0 {
+		pi.Aliases = p.aliases
+	}
+	return pi
+}
+
+// Tests behavior of command tree with plugins remapped (at a plugin level)
+func TestPluginLevelRemapping(t *testing.T) {
+	tests := []struct {
+		test              string
+		pluginVariants    []fakePluginRemapAttributes
+		args              []string
+		expected          []string
+		unexpected        []string
+		expectedFailure   bool
+		activeContextType configtypes.ContextType
+	}{
+		{
+			test: "one unmapped k8s plugin",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:   "dummy",
+					target: configtypes.TargetK8s,
+				},
+			},
+			args:     []string{"dummy", "say", "hello"},
+			expected: []string{"hello"},
+		},
+		{
+			test: "one mapped k8s plugin",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:      "dummy2",
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"dummy"},
+				},
+			},
+			args:     []string{"dummy", "say", "hello"},
+			expected: []string{"hello"},
+		},
+		{
+			test: "once mapped, command using original plugin name is not accessible",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:      "dummy2",
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"dummy"},
+				},
+			},
+			args:            []string{"dummy2"},
+			expectedFailure: true,
+		},
+		{
+			test: "two plugins share aliases shows duplicate warning",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:    "dummy",
+					target:  configtypes.TargetK8s,
+					aliases: []string{"dum"},
+				},
+				fakePluginRemapAttributes{
+					name:      "dummy2",
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"dummy3"},
+					aliases:   []string{"dum"},
+				},
+			},
+			args:     []string{"dummy3", "say", "hello"},
+			expected: []string{"hello", "the alias dum is duplicated across plugins: dummy, dummy3"},
+		},
+		{
+			test: "two plugins sharing aliases does not show duplicate warning if one replaces another",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:    "dummy",
+					target:  configtypes.TargetK8s,
+					aliases: []string{"dum"},
+				},
+				fakePluginRemapAttributes{
+					name:      "dummy2",
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"dummy"},
+					aliases:   []string{"dum"},
+				},
+			},
+			args:       []string{"dummy", "say", "hello"},
+			expected:   []string{"hello"},
+			unexpected: []string{"duplciated across plugins"},
+		},
+		{
+			test: "map to deeper subcommand is valid as long as parent exists",
+			pluginVariants: []fakePluginRemapAttributes{
+				fakePluginRemapAttributes{
+					name:      "dummy2",
+					target:    configtypes.TargetK8s,
+					invokedAs: []string{"dummy", "kubernetes dummy"},
```

Comment:
Could we add a test that only specifies `dummy` in `invokedAs` but that checks that we can still call it as `[]string{"kubernetes", "dummy", "say", "hello"}`?

And the reverse as another test (specify only `kubernetes dummy` but still call as  `[]string{"dummy", "say", "hello"}`?

Reply:
good idea, thanks!